src/test/java/com/ecommerce/stepDefinitions/CartPageSteps.java

```diff
@@ -0,0 +1,99 @@
+			System.out.println("Landed on product details page");
+			ReportManager.getTest().pass("User navigated to the product details page successfully");
+		} catch (AssertionError ae) {
+			System.out.println("Assertion failed" + ae.getMessage());
+			ReportManager.getTest().fail("Failed to land on product details page");
+			throw ae;
+		} catch (Exception e) {
+			System.out.println("An error occured navigating to the product details page " + e.getMessage());
+			ReportManager.getTest().fail("An error occured while navigating to the product details page");
+			throw e;
+		}
+	}
+
+	@When("User click on add to cart button")
+	public void user_click_on_add_to_cart_button() {
+		ProductDetailsPage productDetailsPage = new ProductDetailsPage(driver);
+		productDetailsPage.ClickOnAddProductBtn();
+		ReportManager.getTest().info("Clicked on add to cart button");
+	}
+
+	@Then("User should see item add to cart")
+	public void i_should_see_item_add_to_cart() {
+		ProductDetailsPage procuDetailsPage = new ProductDetailsPage(driver);
+		int ExceptedProductsInShoppingBag = 1;
+		try {
+			Assert.assertEquals(ExceptedProductsInShoppingBag, procuDetailsPage.getProductsInShoppingBag());
+			System.out.println("item added to cart successfully");
+			ReportManager.getTest().pass("item added to cart successfully");
+		} catch (AssertionError ae) {
+			System.out.println("Assertion failed" + ae.getMessage());
+			ReportManager.getTest().fail("Failed to item added to cat");
+			throw ae;
+		} catch (Exception e) {
+			System.out.println("An error occured navigating to the item added to the cart" + e.getMessage());
+			ReportManager.getTest().fail("An error occured navigating to the item added to the cart");
+			throw e;
+		}
+	}
+
+	@When("User click on view cart button")
+	public void user_click_on_view_cart_button() {
+		ProductDetailsPage productDetailsPage = new ProductDetailsPage(driver);
+		productDetailsPage.ClickOnViewCartBtn();
+		ReportManager.getTest().info("Clicked on view cart button");
+	}
+
+	@Then("User navigated to the cart page")
+	public void user_navigated_to_the_cart_page() {
+		CartPage cartPage = new CartPage(driver);
+		String ExpectedCartPageTitle = "My Cart";
+		try {
+			Assert.assertEquals(cartPage.getCartPageTitle(), ExpectedCartPageTitle);
+			System.out.println("Landed on cart page");
+			ReportManager.getTest().pass("Navigated to the cart page successfully");
+		} catch (AssertionError ae) {
+			System.out.println("Assertion failed" + ae.getMessage());
+			ReportManager.getTest().fail("An error occured while navigating to the cart page");
+			throw ae;
+		} catch (Exception e) {
+			System.out.println("An error occured while navigating to the cart page");
+			ReportManager.getTest().fail("An error occured while navigating to the cart page");
+			throw e;
+		}
+	}
+
+	@Then("User should see item in cart")
+	public void user_should_see_item_in_to_cart() {
+		ReportManager.getTest().info("user should see itm in cart");
+
+	}
+
```

Comment:
Where is the Validations here , how can you confirm you are seeing items in the cart

Reply:
I added validation.